packages/ra-ui-materialui/src/button/PrevNextButton.stories.tsx

```diff
@@ -0,0 +1,320 @@
+
+        return {
+            id,
+            first_name,
+            last_name,
+            email,
+            city: address.city(),
+        };
+    }),
+};
+
+const dataProvider = fakeRestDataProvider({
+    ...data,
+    ...{
+        customers: [
+            // add some persisted customers to the data
+            ...data.customers,
+            ...[
+                {
+                    id: 901,
+                    first_name: 'Martin',
+                    last_name: 'McFly',
+                    email: 'Martin.McFly@yahoo.com',
+                    city: 'Hill Valley',
+                },
+                {
+                    id: 902,
+                    first_name: 'Emmett',
+                    last_name: 'Brown',
+                    email: 'Emmett.Brown@yahoo.com',
+                    city: 'Hill Valley',
+                },
+                {
+                    id: 903,
+                    first_name: 'Biff',
+                    last_name: 'Tannen',
+                    email: 'Biff.Tannen@yahoo.com',
+                    city: 'Hill Valley',
+                },
+                {
+                    id: 904,
+                    first_name: 'Clara',
+                    last_name: 'Clayton',
+                    email: 'Clara.Clayton@yahoo.com',
+                    city: 'Hill Valley',
+                },
+            ],
+        ],
+    },
+});
+
+const MyTopToolbar = ({ children }) => (
+    <TopToolbar
+        sx={{
+            justifyContent: 'space-between',
+        }}
+    >
+        {children}
+    </TopToolbar>
+);
+
+const defaultFiedls = [
+    <TextField source="id" key="id" />,
+    <TextField source="first_name" key="first_name" />,
+    <TextField source="last_name" key="last_name" />,
+    <TextField source="email" key="email" />,
+    <TextField source="city" key="city" />,
+];
+
+const defaultInputs = [
+    <TextInput source="first_name" key="first_name" />,
+    <TextInput source="last_name" key="last_name" />,
+    <TextInput source="email" key="email" />,
+    <TextInput source="city" key="city" />,
+];
+
+export const Basic = () => (
+    <MemoryRouter>
+        <AdminContext dataProvider={dataProvider} i18nProvider={i18nProvider}>
+            <AdminUI>
+                <Resource
+                    name="customers"
+                    list={
+                        <List>
+                            <Datagrid rowClick="edit">{defaultFiedls}</Datagrid>
+                        </List>
+                    }
+                    edit={
+                        <Edit
+                            actions={
+                                <MyTopToolbar>
+                                    <PrevNextButton />
+                                    <ShowButton />
+                                </MyTopToolbar>
+                            }
+                        >
+                            <SimpleForm>{defaultInputs}</SimpleForm>
```

Comment:
instead of extracting an array of inputs, extract a custom form component. That's more React-y.